aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -200,77 +123,6 @@ private RESTClient httpClient() {
     return httpClient;
   }
 
-  private AuthSession authSession() {
-    String token = token().get();
-    if (null != token) {
```

Comment:
how will this now work when the token has been provided through the token supplier?

Reply:
The Auth Manager takes care of it. The token supplier returns `properties().get(OAuth2Properties.TOKEN)`. The auth manager will do the same if configured with that property.

Reply:
the default token supplier returned `properties().get(OAuth2Properties.TOKEN)` but what if you had an entirely different supplier that would fetch the token from somewhere else?

Reply:
But how can you create different suppliers? The only way to instantiate this class is through the method `S3V4RestSignerClient.create()`, which is hard-coded in `S3FileIOProperties.applySignerConfiguration()`. And that method does not allow custom token suppliers to be provided.

Reply:
you can always go through `ImmutableS3SignRequest.builder().token(() -> mySecretToken).build()`

Reply:
Hmm I don't get it. You example doesn't compile, the is no `token()` method in `ImmutableS3SignRequest.Builder`.

Reply:
sorry, I meant `ImmutableS3V4RestSignerClient.builder()` (I've updated the above example)

Reply:
Ah OK 👍  

To create a signer this way, you need to:

* Reimplement `S3FileIOProperties` or extend it and override `applySignerConfiguration()`;
* Reimplement `S3FileIO` or extend it and override the constructor and `initialize(Map<String,String)` since `S3FileIOProperties` is hard-coded in these places.

So... I don't think this is a realistic scenario.

Reply:
it actually is a realistic scenario, because that's the API we've been using internally for the Signer

Reply:
*sigh* OK let's put back those methods.